modules/navigation/src/main/java/com/sygic/maps/module/navigation/NavigationFragment.kt

```diff
@@ -29,35 +29,87 @@ import android.util.AttributeSet
 import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
-import androidx.lifecycle.Observer
 import com.sygic.maps.module.common.MapFragmentWrapper
+import com.sygic.maps.module.navigation.component.DISTANCE_UNITS_DEFAULT_VALUE
 import com.sygic.maps.module.navigation.component.PREVIEW_MODE_DEFAULT_VALUE
+import com.sygic.maps.module.navigation.component.SIGNPOST_ENABLED_DEFAULT_VALUE
 import com.sygic.maps.module.navigation.databinding.LayoutNavigationBinding
 import com.sygic.maps.module.navigation.di.DaggerNavigationComponent
 import com.sygic.maps.module.navigation.di.NavigationComponent
 import com.sygic.maps.module.navigation.viewmodel.NavigationFragmentViewModel
+import com.sygic.maps.uikit.viewmodels.common.regional.RegionalManager
+import com.sygic.maps.uikit.viewmodels.common.regional.units.DistanceUnits
+import com.sygic.maps.uikit.viewmodels.navigation.signpost.FullSignpostViewModel
+import com.sygic.maps.uikit.viewmodels.navigation.signpost.SimplifiedSignpostViewModel
 import com.sygic.maps.uikit.views.common.extensions.getBoolean
 import com.sygic.maps.uikit.views.common.extensions.getParcelableValue
-import com.sygic.sdk.map.`object`.MapRoute
+import com.sygic.maps.uikit.views.navigation.signpost.FullSignpostView
+import com.sygic.maps.uikit.views.navigation.signpost.SimplifiedSignpostView
 import com.sygic.sdk.route.RouteInfo
+import javax.inject.Inject
 
 const val NAVIGATION_FRAGMENT_TAG = "navigation_map_fragment_tag"
+internal const val KEY_DISTANCE_UNITS = "distance_units"
+internal const val KEY_SIGNPOST_ENABLED = "signpost_enabled"
 internal const val KEY_PREVIEW_MODE = "preview_mode"
 internal const val KEY_ROUTE_INFO = "route_info"
 
 /**
  * A *[NavigationFragment]* is the core component for any navigation operation. It can be easily used for the navigation
- * purposes. By setting the [routeInfo] object will start the navigation process. Any pre build-in element such as [DirectionInfo]
- * [Signposts], [Infobar], [CurrentSpeed] or [SpeedLimit] may be activated or deactivated and styled.
+ * purposes. By setting the [routeInfo] object will start the navigation process. Any pre build-in element such as
+ * [FullSignpostView], [SimplifiedSignpostView], [Infobar], [CurrentSpeed] or [SpeedLimit] may be activated or deactivated and styled.
  */
 @Suppress("unused", "MemberVisibilityCanBePrivate")
 class NavigationFragment : MapFragmentWrapper<NavigationFragmentViewModel>() {
 
     override lateinit var fragmentViewModel: NavigationFragmentViewModel
+    private lateinit var fullSignpostViewModel: FullSignpostViewModel
```

Comment:
v tomto PR mas este oba VM aj oba view v layoute? Ci budes menit este?

Reply:
jj, vím. ještě jsem se nedostal k tomu ViewStub-u. toto bylo dočasné řešení. v jednom z dalších commutů to bude již upravené.